### DISCUSSION TITLE
JSLT with Apache Camel

### DISCUSSION THREAD

**Discussion by abhishekrongali** [schibsted/jslt#311] [Q&A]
Hi,

We have integrated JSLT with Apache Camel to transform JSON Requests and Response during REST API calls.
Question- We have exchange property in apache camel to hold any values during process, currently we can retrieve the values from exchange property in JSLT (**example** - **$exchange.properties.id**),
but we wanted to know if we can also set the values to exchange property in JSLT itself? 

If can share any sample examples would be really appreciated!

Thanks in advance!!

**Comment by larsga** [comment]
I don't actually understand the question. What is it you want to do? Modify the `$exchange` object inside JSLT? If so, the answer is that JSLT doesn't modify anything. It only produces new values.

I suppose you could plug in an extension function that does this.

**Reply by abhishekrongali** [reply]
Thanks for your response! 
And yeah I wanted to modify the $exchange object inside JSLT! so how can we call the function which is available in class file and pass on the parameters to store it?

**Reply by larsga** [reply]
This guide https://github.com/schibsted/jslt/blob/master/extensions.md explains how to write your own functions in Java.

How to make those available within Camel I'm not sure, tbh.
